src/hooks/v1/RelaunchV1ViaV3Create.ts

```diff
@@ -0,0 +1,148 @@
+          preferClaimed: mod.preferUnstaked,
+          projectId: !mod.projectId?.eq(0)
+            ? mod.projectId?.toString()
+            : undefined,
+        })
+      }
+      return converted
+    }
+    const convertTicketMods = (mods: TicketMod[]) => {
+      const converted: unknown[] = []
+      for (const mod of mods ?? []) {
+        converted.push({
+          percent: splitPercentFrom(
+            parseFloat(permyriadToPercent(mod.percent)),
+          ).toString(),
+          beneficiary: mod.beneficiary,
+          lockedUntil: mod.lockedUntil,
+          preferClaimed: mod.preferUnstaked,
+        })
+      }
+      return converted
+    }
+    return {
+      projectMetadata: projectMetadata
+        ? consolidateMetadata(projectMetadata)
+        : undefined,
+      fundingCycleData: {
+        duration: queuedFC?.duration.mul(SECONDS_IN_DAY).toString(),
+        weight: queuedFC?.weight.toString(),
+        discountRate: discountRateFrom(queuedFC?.discountRate.toString() ?? 0)
+          .div(10)
+          .toString(),
+        ballot: queuedFC?.ballot,
+      },
+      fundingCycleMetadata: {
+        reservedRate: percentToPermyriad(
+          perbicentToPercent(fundingCycleMetadata?.reservedRate),
+        ).toString(),
+
+        redemptionRate: percentToPermyriad(
+          perbicentToPercent(fundingCycleMetadata?.bondingCurveRate),
+        ).toString(),
+        ballotRedemptionRate: percentToPermyriad(
+          perbicentToPercent(fundingCycleMetadata?.bondingCurveRate),
+        ).toString(),
+        pausePay: fundingCycleMetadata?.payIsPaused,
+        // pauseDistribution: , // Not in v1
+        // pauseRedeem: , // Not in v1
+        allowMinting: fundingCycleMetadata?.ticketPrintingIsAllowed,
+        // allowTerminalMigration: , // Not in v1
+        // allowControllerMigration: , // Not in v1
+        // holdFees: ,// Not in v1
+        // useTotalOverflowForRedemptions: ,// Not in v1
+        // useDataSourceForPay: ,// Not in v1
+        // useDataSourceForRedeem: ,// Not in v1
+        // dataSource: ,// Not in v1
+        // preferClaimedTokenOverride: ,// Not in v1
+        // metadata: ,// Not in v1
+      },
+      fundAccessConstraints: [
+        {
+          distributionLimit: fromWad(queuedFC?.target),
+          distributionLimitCurrency: queuedFC?.currency.toString(),
+          overflowAllowance: '0', // nothing
+          overflowAllowanceCurrency: '0', // nothing
+        },
+      ],
+      payoutGroupedSplits: {
+        group: 1,
+        splits: queuedPayoutMods ? convertPayoutMods(queuedPayoutMods) : [],
+      },
+      reservedTokensGroupedSplits: {
+        group: 1,
+        splits: queuedTicketMods ? convertTicketMods(queuedTicketMods) : [],
+      },
+    }
+  }, [
+    fundingCycleMetadata?.bondingCurveRate,
+    fundingCycleMetadata?.payIsPaused,
+    fundingCycleMetadata?.reservedRate,
+    fundingCycleMetadata?.ticketPrintingIsAllowed,
+    projectMetadata,
```

Comment:
**question** I wonder if memoization might be slightly unsafe here. For example If some key within projectMetadata changes, the `useMemo` won't re-evaluate when it probably should

Reply:
maybe im missing why it wouldn't, why wouldn't it 😅

Reply:
I think due to the nature of the way that useContext works the object being returned would be a 'new' object when any of the properties change so the dep would fire the callback

Reply:
I think due to the nature of the way that useContext works the object being returned would be a 'new' object when any of the properties change so the dep would fire the callback